rts/motoko-rts/src/gc/generational.rs

```diff
@@ -117,7 +117,8 @@ static mut OLD_GENERATION_THRESHOLD: usize = 32 * 1024 * 1024;
 static mut PASSED_CRITICAL_LIMIT: bool = false;
 
 #[cfg(feature = "ic")]
-const CRITICAL_MEMORY_LIMIT: usize = (4096 - 512) * 1024 * 1024 - crate::memory::MEMORY_RESERVE;
+const CRITICAL_MEMORY_LIMIT: usize =
+    (4096 - 512) * 1024 * 1024 - crate::memory::GENERAL_MEMORY_RESERVE;
```

Comment:
```suggestion
    (4096 - 512) * 1024 * 1024 - crate::memory::GENERAL_MEMORY_RESERVE;
```
Nit: Maybe use your new constants here too? And it definition of OLD_GENERATION_THRESHOLD above?

Reply:
I looked into this and am not sure whether I should reuse the same or similar heuristics as in incremental GC. One aspect is that the GC does not need much extra space except for the mark bitmap (128MB). Moreover, there is no reclamation delay by multiple increments but the free space is entirely collected when the limit is exceeded and/or full collection is initiated. This is why I believe we could keep the more generous limit. But most probably, the full collection would not fit into instruction limit anyways for larger fine-grained heaps. 
I am interested in your thoughts. Maybe there are an additional aspects I should consider?

Reply:
Oh all I was suggesting is maybe replace     1024 by MB etc, as elsewhere. Totally optional